### ISSUE TITLE
concatenate() fails for some `met` files

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#2213]
I've tried the example from `concatenate,oce-method`, and it works, but the following code (which is functionally identical), doesn't:

``` r
library(oce)
#> Loading required package: gsw
metfile <- download.met(43405, 2023, 12, 'hour', type='csv')
met1 <- read.met(metfile)
metfile <- download.met(43405, 2024, 1, 'hour', type='csv')
met2 <- read.met(metfile)
met <- concatenate(met1, met2)
#> Error in `[[<-.data.frame`(`*tmp*`, n, value = c("2023-12-01 00:00", "2023-12-01 01:00", : replacement has 1488 rows, data has 744
```

<sup>Created on 2024-05-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Any ideas what might be different here? Or is it an ID10T error that I can't see?

**Comment by richardsc** [comment]
Ok, I tried removing the `type="csv"` argument from my `download.met()` calls, and that seemed to fix it. 

Seems to me like it should work regardless of what kind of file the `met` object was created from though, so I'll leave this open for @dankelley to have a look at (though, I know he's busy until Tuesday with other things, so please note this is no rush).

**Comment by dankelley** [comment]
I'll take a look.  I spent the weekend on that Tuesday task, so I can relax on that.

**Comment by dankelley** [comment]
Oh, I see.  It's being stored as a data frame.  It is illegal to lengthen one column of a data frame without lengthening all of them.  The `data` slot of non-`csv` cases must be getting stored as `list` objects.  I could detect whether it's a data frame first, then convert to a list, and convert back to a data frame at the end.  But this may be the only oce object that stores the `data` slot as a data frame, so I think the more sensible scheme would be to cause the `read.met()` to return a list.

Maybe @richardsc and I can chat about this today, if he will be at Dal.  I don't really think many users will be storing `rda` files of old met data, so I don't see much harm in changing the format.  (Frankly, the data source changes so frequently that I think the downloading software needs updating every year or so, making met data a bit of a pain ... hence the creation of github.com/dankelley/doc)

**Comment by dankelley** [comment]
I coded a fix (not pushed to GH) and made a test file in oce-issues repo, file oce-issues/22xx/2213/2213a.R and it makes a graph as below.

So, the code no longer fails, but why that gap?  I'll need to look into this some more.


![2213a](https://github.com/dankelley/oce/assets/99469/d9d3d7b0-c9c4-44a8-bafe-28bb9b1d342f)

**Comment by dankelley** [comment]
Oh, hang on.  That gap is *in the data files* (see below), so ignore my worry on the gap.

I think I've run into these gaps before, actually, when trying to make things to show class.  (I guess it would be better to say "anybody have an idea on the gap" and maybe the met students would say "yeah, our prof told us never to trust those data" ... or maybe I'm being too Panglossian.)

<img width="1102" alt="Screenshot 2024-05-06 at 7 23 03 AM" src="https://github.com/dankelley/oce/assets/99469/51722452-85e6-49f6-87d2-5e7c2f9f6ec3">

**Comment by richardsc** [comment]
Yeah, I think the missing data was just a station problem. 😄 

I will be around Dal this morning, and will pop by!

**Comment by dankelley** [comment]
I think this is solved in "develop" commit fe7d581e0dc631a99b4ab3873919f15a440ca332

**Comment by dankelley** [comment]
Dear reporter,

Do you think this issue (as defined by its title) has been addressed? If so, please close it. If not, please add a comment explaining what remains to be done.

We use open issues as a sort of "to do" list for the project.

*Thanks!*

PS.  This is a standardized reply.
